### ISSUE TITLE
Answer for 'Functional Programming: Sort an Array Alphabetically using the sort Method' doesn't work in Chrome

### ISSUE THREAD

**Issue by guayom** [freeCodeCamp/freeCodeCamp#35823]
**Describe your problem and how to reproduce it:**

I was doing the [Functional Programming: Sort an Array Alphabetically using the sort Method](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/functional-programming/sort-an-array-alphabetically-using-the-sort-method) challenge, in which you're supposed to sort an array of letters alphabetically. So I tried what I thought was the correct answer. It didn't work. I looked at the answer and it was exactly the same but with a longer syntax:
```javascript
return arr.sort((a,b)=> a>b));
```
I was very confused, I was so confident about that answer! So I started trying different things in codepen and jsfiddle, but that method wasn't working at all:
```javascript
// Chrome output
const arr = ["c", "a", "b"];
console.log(arr.sort((a,b)=>a>b)); //["c", "a", "b"]
console.log(arr.sort((a, b) => a.localeCompare(b))); //["a", "b", "c"]
console.log(arr.sort()); //["a", "b", "c"]
```
Still very confused I opened Firefox and did the same thing. But this time, all methods worked.
```javascript
// Firefox output
const arr = ["c", "a", "b"];
console.log(arr.sort((a,b)=>a>b)); //["a", "b", "c"]
console.log(arr.sort((a, b) => a.localeCompare(b))); //["a", "b", "c"]
console.log(arr.sort()); //["a", "b", "c"]
```
It appears that the latest versions of Chrome won't let you sort arrays alphabetically that way anymore. But I don't know what would be the right way to tackle that problem.

Probably a good first step is to add a disclaimer to the solution guide explaining the issue.

What do you guys think?

-- EDIT:
It may be obvious, but I should mention that this is a problem for all the people using later versions of Chrome, because in both the challenge explanation and the solution, the method that is used to sort the array won't work. So even if you copy and paste the solution, the tests will fail.

**Add a Link to the page with the problem:**
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/functional-programming/sort-an-array-alphabetically-using-the-sort-method

**Tell us about your browser and operating system:**

* Browser Name: Chrome
* Browser Version: 73.0.3683.86 (Official Build) (64-bit)
--
* Browser Name: Firefox
* Browser Version: 67.0b8 (64-bit)
--
* Operating System: macOS Mojave 10.14.3

**Comment by ojeytonwilliams** [comment]
What's going on here is that Chrome is being strict about the spec while Firefox is more relaxed.  From [here](https://www.ecma-international.org/ecma-262/6.0/#sec-array.prototype.sort):
> If comparefn is not undefined, it should be a function that accepts two arguments x and y and returns a negative value if x < y, zero if x = y, or a positive value if x > y.

The instructions are a bit different on [rocks](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/functional-programming/sort-an-array-alphabetically-using-the-sort-method/), as the misleading example has been replaced with correct code.  Do you think the new instructions do a good enough job conveying how to use `sort`?

**Comment by guayom** [comment]
Thank you for the explanation! Now I understand what the problem is.
I think that the instructions are very good at explaining how `sort()` works.
But the problem remains. In chrome, you won't be able to sort alphabetically an array of letters by using `arr.sort((a,b)=>a>b)`., as the example shows.

Actually, if you copy and paste the second example from the instructions, you will get a different result. I think that would be confusing for people that is using chrome and is just learning js.

Maybe it would be good to have 2 different challenges? One to sort numbers and one alphabetically. And in the one for alphabetical sorting, explain this issue and show another method?

**Comment by ojeytonwilliams** [comment]
@guayom Ah, my bad, I linked to the wrong page!  I should have pointed you to [here](https://www.freecodecamp.rocks/learn/javascript-algorithms-and-data-structures/functional-programming/sort-an-array-alphabetically-using-the-sort-method).  I've edited my original comment, but I thought I'd best add this so you know what happened. 

I think it's okay to keep one challenge, but it might be worth modifying this one a little.  For example, the first line could read something like 

> The `sort` method sorts the elements of an array and returns it.  `sort` can be called with an optional callback function that takes two arguments and returns -1, 0 or 1 if they are in order, equal or out of order.

I think it's best to avoid boolean comparisons entirely.  As you've noticed, they are not reliable!
